### DISCUSSION TITLE
Did anyone notice problems with receiving serial data?

### DISCUSSION THREAD

**Discussion by SpenceKonde** [SpenceKonde/DxCore#287] [Q&A]
It looks to me like the inline assembly used to improve performance and reduce flash size is accessing the current HEAD at the wrong offset, and god only knows what value it's reading. thought it writes to the correct location. We read it from Z+17 but write to Z+ 19 (the latter is consistent with everything else. This would appear to break all serial RX. Has anyone else noticed anomalies here? Trivial fix, but my mind is blown that nobody is screaming at me over this!

**Comment by MX682X** [comment]
It might be worth to double check f this problem had something to do with it:
https://github.com/SpenceKonde/megaTinyCore/issues/692

**Reply by SpenceKonde** [reply]
Oh it could be! Good thinking!
